### PR TITLE
Preliminary cleanup of local exceptions in Ruby

### PR DESCRIPTION
**Pull request by bernardnormier**
This is a first cleanup of local exceptions in Ruby. Most of the generated code was not used.

### CODE REVIEW

File: ruby/ruby/Ice.rb [hunk 1]
```diff
@@ -233,20 +233,20 @@ def FormatType.each(&block)
 #
 require 'Ice/BuiltinSequences.rb'
 require 'Ice/Context.rb'
-require 'IceLocal/Current.rb'
-require 'IceLocal/Communicator.rb'
-require 'IceLocal/EndpointSelectionType.rb'
+require_relative 'IceLocal/Current.rb'
```
Comment (bernardnormier) [inline ruby/ruby/Ice.rb:236]: We should also use require_relative for the Ice/ generated files. For a follow-up PR.

File: ruby/ruby/IceLocal/LocalException.rb [hunk 2]
```diff
@@ -3,635 +3,454 @@
...
+        def initialize(reason = '')
+            @reason = reason
+        end
 
-            attr_accessor :reason
+        def to_s
+            '::Ice::InitializationException'
         end
 
-        T_InitializationException = ::Ice::__defineException('::Ice::InitializationException', InitializationException, nil, [["reason", ::Ice::T_string, false, 0]])
```
Comment (bernardnormier) [inline ruby/ruby/IceLocal/LocalException.rb:26]: All this code is for user exceptions. We didn't use it for local exceptions!

**Review by externl — APPROVED** [review]

File: ruby/ruby/IceLocal/LocalException.rb [hunk 3]
```diff
@@ -3,635 +3,454 @@
...
-        class CFNetworkException < ::Ice::SocketException
-            def initialize(error=0, domain='')
-                super(error)
-                @domain = domain
-            end
-
-            def to_s
-                '::Ice::CFNetworkException'
-            end
+    class CFNetworkException < ::Ice::SocketException
```
Comment (pepone) [inline ruby/ruby/IceLocal/LocalException.rb:247]: isn't this only for iOS, can we remove it?

**Review by pepone — APPROVED** [review]

File: ruby/ruby/IceLocal/LocalException.rb (hunk 3 above)
Comment (bernardnormier) [reply ruby/ruby/IceLocal/LocalException.rb:247]: Yes, I'll remove it in the refactoring PR.

**Pull request merged**